File: Software/Java/src/com/dexterind/gopigo/utils/Commands.java
/*
 * #%L
 * **********************************************************************
 * ORGANIZATION  :  DexterIndustries
 * PROJECT       :  GoPiGo Java Library
 * FILENAME      :  Gopigo.java
 * AUTHOR        :  Marcello Barile <marcello.barile@gmail.com>
 *
 * This file is part of the GoPiGo Java Library project. More information about
 * this project can be found here:  https://github.com/DexterInd/GoPiGo
 * **********************************************************************
 * %%
 * This project is open source. These files have been made available
 * online through a Creative Commons Attribution-ShareAlike 3.0 license.
 *
 *      http://creativecommons.org/licenses/by-sa/3.0/
 *
 * #L%
 */
package com.dexterind.gopigo.utils;

public class Commands {
  /**
   * Move forward with PID.
   */
  public static final int FWD                  = 119;
  /**
   * Move forward without PID.
   */
  public static final int MOTOR_FWD            = 105;
  /**
   * Move backward with PID.
   */
  public static final int BWD                  = 115;
  /**
   * Move backward without PID.
   */
  public static final int MOTOR_BWD            = 107;
  /**
   * Turn left by turning off one motor.
   */
  public static final int LEFT                 = 97;
  /**
   * Rotate left by running both motors in opposite direction.
   */
  public static final int LEFT_ROT             = 98;
  /**
   * Turn right by turning off one motor.
   */
  public static final int RIGHT                = 100;
  /**
   * Rotate right by running both motors in opposite direction.
   */
  public static final int RIGHT_ROT            = 110;
  /**
   * Stop the GoPiGo.
   */
  public static final int STOP                 = 120;
  /**
   * Increase the speed by 10.
   */
  public static final int ISPD                 = 116;
  /**
   * Decrease the speed by 10.
   */
  public static final int DSPD                 = 103;
  /**
   * Control motor1.
   */
  public static final int M1                   = 111;
  /**
   * Control motor2.
   */
  public static final int M2                   = 112;
  /**
   * Read the voltage of the batteries.
   */
  public static final int VOLT                 = 118;
  /**
   * Read the distance from the ultrasonic sensor.
   */
  public static final int US                   = 117;
  /**
   * Turn ON/OFF the LEDs.
   */
  public static final int LED                  = 108;
  /**
   * Rotate the servo.
   */
  public static final int SERVO                = 101;
  /**
   * Set the encoder targeting.
   */
  public static final int ENC_TGT              = 50;
  /**
   * Read the firmware version.
   */
  public static final int FW_VER               = 20;
  /**
   * Enable the encoders.
   */
  public static final int EN_ENC               = 51;
  /**
   * Disable the encoders.
   */
  public static final int DIS_ENC              = 52;
  /**
   * Read the encoder status.
   */
  public static final int READ_ENC_STATUS      = 53;
  /**
   * Enable the servo.
   */
  public static final int EN_SERVO             = 61;
  /**
   * Disable the servo.
   */
  public static final int DIS_SERVO            = 60;
  /**
   * Set the speed of the right motor.
   */
  public static final int SET_LEFT_SPEED       = 70;
  /**
   * Set the speed of the left motor.
   */
  public static final int SET_RIGHT_SPEED      = 71;
  /**
   * Enable the communication timeout.
   */
  public static final int EN_COM_TIMEOUT       = 80;
  /**
   * Disable the communication timeout.
   */
  public static final int DIS_COM_TIMEOUT      = 81;
  /**
   * Read the timeout status.
   */
  public static final int TIMEOUT_STATUS       = 82;
  /**
   * Read the encoder values.
   */
  public static final int ENC_READ             = 53;
  /**
   * Test the trim values.
   */
  public static final int TRIM_TEST            = 30;
  /**
   * Write the trim values.
   */
  public static final int TRIM_WRITE           = 31;
  /**
   * Read the trim values.
   */
  public static final int TRIM_READ            = 32;
  /**
   * Digital write on a port.
   */
  public static final int DIGITAL_WRITE        = 12;
  /**
   * Digital read on a port.
   */
  public static final int DIGITAL_READ         = 13;
  /**
   * Analog read on a port.
   */
  public static final int ANALOG_READ          = 14;
  /**
   * Analog read on a port.
   */
  public static final int ANALOG_WRITE         = 15;
  /**
   * Set up the pin mode on a port.
   */
  public static final int PIN_MODE             = 16;

  /**
   * This allows us to be more specific about which commands contain unused ints
   */
  public static final int UNUSED               = 0;

  public Commands(){}
}

File: Software/NodeJS/libs/commands.js
module.exports = {
    fwd                                         : [119]   // Move forward with PID
  , motor_fwd                                   : [105]   // Move forward without PID
  , bwd                                         : [115]   // Move back with PID
  , motor_bwd                                   : [107]   // Move back without PID
  , left                                        : [97]    // Turn Left by turning off one motor
  , left_rot                                    : [98]    // Rotate left by running both motors is opposite direction
  , right                                       : [100]   // Turn Right by turning off one motor
  , right_rot                                   : [110]   // Rotate Right by running both motors is opposite direction
  , stop                                        : [120]   // Stop the GoPiGo
  , ispd                                        : [116]   // Increase the speed by 10
  , dspd                                        : [103]   // Decrease the speed by 10
  , m1                                          : [111]   // Control motor1
  , m2                                          : [112]   // Control motor2

  , volt                                        : [118]   // Read the voltage of the batteries
  , us                                          : [117]   // Read the distance from the ultrasonic sensor
  , led                                         : [108]   // Turn On/Off the LED's
  , servo                                       : [101]   // Rotate the servo
  , enc_tgt                                     : [50]    // Set the encoder targeting
  , fw_ver                                      : [20]    // Read the firmware version
  , en_enc                                      : [51]    // Enable the encoders
  , dis_enc                                     : [52]    // Disable the encoders
  , read_enc_status                             : [53]    // Read encoder status
  , en_servo                                    : [61]    // Enable the servo's
  , dis_servo                                   : [60]    // Disable the servo's
  , set_left_speed                              : [70]    // Set the speed of the right motor
  , set_right_speed                             : [71]    // Set the speed of the left motor
  , en_com_timeout                              : [80]    // Enable communication timeout
  , dis_com_timeout                             : [81]    // Disable communication timeout
  , timeout_status                              : [82]    // Read the timeout status
  , enc_read                                    : [53]    // Read encoder values
  , trim_test                                   : [30]    // Test the trim values
  , trim_write                                  : [31]    // Write the trim values
  , trim_read                                   : [32]

  , dWrite                                      : [12]    // Digital write on a port
  , dRead                                       : [13]    // Digital read on a port
  , aRead                                       : [14]    // Analog read on a port
  , aWrite                                      : [15]    // Analog read on a port
  , pMode                                       : [16]    // Set up the pin mode on a port

  // Grove IR sensor
  // Read the button from IR sensor
  , irRead                                      : [21]
  // Set pin for the IR reciever
  , irRecvPin                                   : [22]

  // This allows us to be more specific about which commands contain unused bytes
  , unused                                      : 0
};
